Makefile

```diff
@@ -22,7 +23,7 @@ dir:
 $(exec): $(objs)
 	$(CXX) $(CXXFLAGS) $(objs) -o $(exec) $(LDFLAGS)
 	
-$(objs_dir)/%.o: $(src_dir)/%.cc
+$(objs_dir)/%.o: $(src_dir)/%.cc $(src_dir)/%.h
```

Comment:
This seems not necessary.

Reply:
If not, the change to the .h file seems will not trigger the recompilation.